### PR TITLE
sc_vfs: fix _ls_handler

### PR DESCRIPTION
**Pull request by vincent-d**
Calls to vfs_closedir were missing at the end of _ls_handler causing umount
ro return an error.

### CODE REVIEW

File: sys/shell/commands/sc_vfs.c [hunk 1]
```diff
@@ -388,7 +390,15 @@ int _ls_handler(int argc, char **argv)
         ++nfiles;
     }
     printf("total %u files\n", nfiles);
-    return 0;
+
+close_and_end:
```
Comment (gebart) [inline sys/shell/commands/sc_vfs.c]: If you change the above part to  
```c
do {
... 
} while (0);
```
you can avoid the `goto`

Comment (vincent-d) [inline sys/shell/commands/sc_vfs.c]: I'm not sure to see how. But this should work as well:

```C
    while (1) {
        vfs_dirent_t entry;
        res = vfs_readdir(&dir, &entry);
        if (res < 0) {
            _errno_string(res, (char *)buf, sizeof(buf));
            printf("vfs_readdir error: %s\n", buf);
            if (res == -EAGAIN) {
                /* try again */
                continue;
            }
            ret = 2;
            break;
        }
        if (res == 0) {
            /* end of stream */
            break;
        }
        printf("%s\n", entry.d_name);
        ++nfiles;
    }
    if (ret == 0) {
        printf("total %u files\n", nfiles);
    }
    res = vfs_closedir(&dir);
    if (res < 0) {
        _errno_string(res, (char *)buf, sizeof(buf));
        printf("vfs_closedir error: %s\n", buf);
        return 2;
    }
    return ret;
```

Comment (miri64) [inline sys/shell/commands/sc_vfs.c]: I'd prefer that.

**Comment by vincent-d** [comment]
Removed goto

**Comment by vincent-d** [comment]
All green, let's go

**Pull request merged**